### PR TITLE
feat(content): Add the Mammoth (Pirate Behemoth)

### PR DESCRIPTION
**Pull request by 1010todd**
**Content (Artwork / Ship)**

battle behemoth i24 thumb@2x

## Summary
Add a modified variant of Behemoth, Mammoth. It have unique sprite and stats rarely found in Large Northern Pirate fleet (as rare as Bactrian) after Battle of Alioth during FW.
Used in FW Bounty 2 as replacement for "Behemoth (Speedy)".
Also intended to be used for [Unfettered Campaign](https://github.com/endless-sky/endless-sky/pull/6416) Mammoth in one mission. (Specifically [Unfettered 1, L692](https://github.com/endless-sky/endless-sky/pull/6416/files#diff-41c62890e3174f734dfa4746e5b4d932264f6264f559210f8ea4cb5759d6904eR692))

<details>
<summary>Extra Images</summary>
Top-down

battle behemoth i24@2x

In-game (Old loadout)

mammoth ingame

Sideview

mammoth side@2x

Flying scene

mammoth fly1@10x

</details>

## Save File
This is the save file right before fighting Moonless Night (FW Bounty 2), the bounty target is in Eltanin 2 jumps away. Provided is a Clipper with Meteor Missiles and Heavy Rockets, one parked Osprey, and one parked Falcon.
[!FWBt2 Test~~BeforeFight.txt](https://github.com/endless-sky/endless-sky/files/10368288/FWBt2.Test.BeforeFight.txt)

## Testing Done
Play tested the affected mission, still easily doable in a reasonable ship player might acquire as long as they follow the advice given.

## Artwork Checklist
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [x] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: https://github.com/endless-sky/endless-sky-assets/pull/113
 - [x] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/335

### CODE REVIEW

File: data/human/ships.txt [hunk 1]
```diff
@@ -555,6 +555,73 @@ ship "Behemoth"
...
+	turret 14.5 0
+	leak "leak" 30 50
+	leak "flame" 30 80
+	leak "big leak" 50 30
+	explode "small explosion" 15
+	explode "medium explosion" 25
+	explode "large explosion" 35
+	explode "huge explosion" 20
+	"final explode" "final explosion large"
+	description "Rare extensively modified Behemoth used by some pirate and smugglers to transport large amount of goods. It is well armed enough to handle even a Navy Frigate if caught and fast enough to dash away from anything bigger."
```
Comment (McloughlinGuy) [inline data/human/ships.txt:621]:
```suggestion
	description "This is a rare extensively modified Behemoth used by some pirate and smugglers to transport large amounts of goods. It is well armed enough to handle even a Navy Frigate if caught and fast enough to dash away from anything bigger."
```

File: data/human/ships.txt [hunk 2]
```diff
@@ -555,6 +555,73 @@ ship "Behemoth"
...
+	turret 14.5 0
+	leak "leak" 30 50
+	leak "flame" 30 80
+	leak "big leak" 50 30
+	explode "small explosion" 15
+	explode "medium explosion" 25
+	explode "large explosion" 35
+	explode "huge explosion" 20
+	"final explode" "final explosion large"
+	description "This is a rare, extensively modified Behemoth used by some pirate and smugglers to transport large amounts of goods. It is well armed enough to handle even a Navy Frigate if caught, and fast enough to dash away from anything bigger."
```
Comment (EjoThims) [inline data/human/ships.txt:621]:
```suggestion
	description "This is a rare, extensively modified Behemoth used by some pirate and smugglers to transport large amounts of goods. It is well armed enough to handle even a Navy Frigate if caught and fast enough to dash away from anything bigger."
```

I was just commenting on this lol

Correct without the comma here; that's not a new sentence.

**Review by EjoThims — CHANGES_REQUESTED** [review]

**Comment by Saugia** [comment]
I believe, to keep consistency, instead of labelling it "modified", you should do the same thing you did with the Scrapper and give it a unique pirate name. Modified ships aren't necessarily pirates, and I would like to see more pirate themed ships with respective names similar to the scrapper.

**Comment by Saugia** [comment]
I'd also like to see this Behemoth be the one used in the FW mission that requires you to hunt one down.

**Comment by Saugia** [comment]
After discussing it in discord, this is where I'd like to see this Behemoth go:

The FW one can be the model Becca made, and rather than being a straight up Pirate Behemoth as I'd like to see the one here being, it'd be a stolen Behemoth that the pirates briefly messed with.

The Behemoth here can be a rare spawn that starts to appear a bit later into the story (human story wise), similar to how the navy gets different weapons later, so have an event defining the usage of these ships somewhere mid-ish in FW story. Instead of it being called a Behemoth, I'd like to see it use the same naming convention as the Scrapper in #8015, which truly puts it in a place as a Pirate ship.

**Comment by Saugia** [comment]
Mammoth seems like a good name.

File: data/human/ships.txt [hunk 3]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+		"weapon capacity" 288
+		"engine capacity" 129
+		"scan interference" 2
+		weapon
+			"blast radius" 140
+			"shield damage" 1400
+			"hull damage" 780
+			"hit force" 2100
+	outfits
+		"Sidewinder Launcher" 2
```
Comment (Saugia) [inline data/human/ships.txt:2321]:
```suggestion
		"Sidewinder Missile Launcher" 2
```

Comment (Saugia) [reply data/human/ships.txt:2321]: Whoops.

File: data/human/ships.txt [hunk 4]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+		"Caldera Afterburner"
+		"Scram Drive"
+		
+	engine 17 147 0.8
+	engine -17 147 0.8
+	engine -34 143 0.8
+	engine 34 143 0.8
+	engine -56 148.5 0.6
+	engine 56 148.5 0.6
+	gun -47.5 -57.5
```
Comment (Saugia) [inline data/human/ships.txt:2344]:
```suggestion
	gun -47.5 -57.5 "Sidewinder Missile Launcher"
```

File: data/human/ships.txt [hunk 5]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	gun 47.5 -57.5
```
Comment (Saugia) [inline data/human/ships.txt:2345]:
```suggestion
	gun 47.5 -57.5  "Sidewinder Missile Launcher"
```

File: data/human/ships.txt [hunk 6]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	turret -31 -50 "Heavy Anti-Missile Turret"
+	turret 31 -50 "Heavy Anti-Missile Turret"
+	turret -47.5 33.5
```
Comment (Saugia) [inline data/human/ships.txt:2348]:
```suggestion
	turret -47.5 33.5 "Proton Turret"
```

File: data/human/ships.txt [hunk 7]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	turret 47.5 33.5
```
Comment (Saugia) [inline data/human/ships.txt:2349]:
```suggestion
	turret 47.5 33.5 "Proton Turret"
```

File: data/human/ships.txt [hunk 8]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	turret -22 66.5 "Heavy Anti-Missile Turret"
+	turret 22 66.5 "Heavy Anti-Missile Turret"
+	turret -14.5 0
```
Comment (Saugia) [inline data/human/ships.txt:2352]:
```suggestion
	turret -14.5 0 "Proton Turret"
```

File: data/human/ships.txt [hunk 9]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	gun -47.5 -57.5
+	gun 47.5 -57.5
+	turret -31 -50 "Heavy Anti-Missile Turret"
+	turret 31 -50 "Heavy Anti-Missile Turret"
+	turret -47.5 33.5
+	turret 47.5 33.5
+	turret -22 66.5 "Heavy Anti-Missile Turret"
+	turret 22 66.5 "Heavy Anti-Missile Turret"
+	turret -14.5 0 
+	turret 14.5 0
```
Comment (Saugia) [inline data/human/ships.txt:2353]:
```suggestion
	turret 14.5 0 "Proton Turret"
```

**Review by Saugia — CHANGES_REQUESTED** [review]
We tend to define weapons on hardpoints even if the game places them there without doing so.

**Comment by Saugia** [comment]
I know there are Behemoths that use Proton turrets, but I forget, are those ever found in the south/southwest? I don't think Protons are generally found in the south anyways. If they're more for syndicate space, the Mammoth makes more sense get different turrets there. Though I'd like more opinions on this first.

File: data/human/ships.txt (hunk 4 above)
Comment (1010todd) [reply data/human/ships.txt:2344]: This is unnecessary especially if there's only single outfit to fit in the ports.

File: data/human/ships.txt [hunk 10]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+	turret 14.5 0 "Proton Turret"
+	leak "leak" 30 50
+	leak "flame" 30 80
+	leak "big leak" 50 30
+	explode "small explosion" 15
+	explode "medium explosion" 25
+	explode "large explosion" 35
+	explode "huge explosion" 20
+	"final explode" "final explosion large"
+	description "This is a rare, extensively modified Behemoth used by some pirates and smugglers to transport large amounts of goods. It is well armed enough to handle even a Navy Frigate if caught and fast enough to dash away from anything bigger."
```
Comment (tibetiroka) [inline data/human/ships.txt:2362]:
```suggestion
	description "This is a rare, extensively modified Behemoth used by some pirates and smugglers to transport large amounts of goods. It is armed well enough to handle even a Navy Frigate if caught and fast enough to dash away from anything bigger."
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]

File: data/human/ships.txt (hunk 4 above)
Comment (Hurleveur) [reply data/human/ships.txt:2344]: Yeah I never do this either

File: data/human/free worlds 2 middle.txt [hunk 11]
```diff
@@ -2020,6 +2020,21 @@ event "end battle of alioth"
 
 
 
+mission "Pirate Use Mammoth"
+	landing
+	invisible
+	to offer
+		has "event: end battle of alioth"
+	on offer
+		event "pirate use mammoth"
```
Comment (Hurleveur) [inline data/human/free worlds 2 middle.txt:2029]:
```suggestion
		event "pirate use mammoth"
		fail
```
make it fail itself to not have missions spam in the savefile

File: data/human/ships.txt [hunk 12]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+		"Sidewinder Missile" 90
+		"Proton Turret" 4
+		"Heavy Anti-Missile Turret" 4
+		
+		"Fission Reactor"
+		"LP144a Battery Pack"
+		"D41-HY Shield Generator"
+		"Large Radar Jammer"
+		"Cooling Ducts"
+		"Laser Rifle" 12
```
Comment (Hurleveur) [inline data/human/ships.txt:2331]:
```suggestion
		"Laser Rifle" 18
```

File: data/human/ships.txt [hunk 13]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+		
+		"A370 Atomic Thruster"
+		"A375 Atomic Steering"
+		"Caldera Afterburner"
+		"Scram Drive"
```
Comment (Hurleveur) [inline data/human/ships.txt:2336]:
```suggestion
		"Scram Drive"
		"Catalytic Ramscoop"
```

File: data/human/ships.txt [hunk 14]
```diff
@@ -2293,6 +2293,76 @@ ship "Leviathan"
...
+		weapon
+			"blast radius" 140
+			"shield damage" 1400
+			"hull damage" 780
+			"hit force" 2100
+	outfits
+		"Sidewinder Missile Launcher" 2
+		"Sidewinder Missile" 90
+		"Proton Turret" 4
+		"Heavy Anti-Missile Turret" 4
```
Comment (Hurleveur) [inline data/human/ships.txt:2324]:
```suggestion
		"Anti-Missile Turret" 4
```

File: data/human/free worlds 2 middle.txt [hunk 15]
```diff
@@ -2020,6 +2020,21 @@ event "end battle of alioth"
...
+
+event "pirate use mammoth"
+	fleet "Large Northern Pirates"
+		add variant 1
```
Comment (Hurleveur) [inline data/human/free worlds 2 middle.txt:2033]:
```suggestion
		add variant
```
its always 1 by default no need to specify it

**Review by Hurleveur — CHANGES_REQUESTED** [review]
Nice that the name from the Unfettered campaign is used, this one having big fuel makes a lot of sense, really liking this ship def gonna be my flagship!

That said, whats the deal with it lacking 36 outfit space and 8 weapon space? It could have outfit expansions for the outfit spaces but that'd be weird and I think it'd make more sense if it was built in.
To fix both issues I think we should replace all heavy anti missiles turrets by light versions, making plenty of space (14 * 4 = 56, less the 36 extra meaning we would have 20 of wiggle room, allowing us to equip a catalytical ramscoop.

You say its 9 jumps away but we only have fuel for 150 * 6 jumps when using a scram drive. That said, with a catalytic ramscoop you'd be able to go as far as you need to.

I'm not a content reviewer though so not demanding changes, just suggesting them. I love to optimize builds.

**Comment by 1010todd** [comment]
> That said, whats the deal with it lacking 36 outfit space and 8 weapon space? 
Seems like I didn't have Sidewinders in my test version so it apparently got enough space on my local.

File: data/human/ships.txt (hunk 4 above)
Comment (Saugia) [reply data/human/ships.txt:2345]: I suppose that is fair. If it was using any other gun mounts I'd say we should at that point.

**Comment by Saugia** [comment]
@warp-core Alright Mr. Balance Man, your turn.

**Comment by 1010todd** [comment]
Play tested the FW Bounty 2 mission, easily doable with missile clipper (2 Meteor, 2 Heavy Rockets). Exact weapon doesn't really matter much, meteor seems to be enough to strip it down to half hull.

image

File: data/human/free worlds 0 prologue.txt [hunk 16]
```diff
@@ -1068,6 +1068,42 @@ mission "FW Bounty 1"
...
+		"Scram Drive"
+
+	engine 
+	engine 
+	engine 
+	engine 
+	engine 
+	engine 
+	gun 
+	gun
```
Comment (warp-core) [inline data/human/free worlds 0 prologue.txt:1095]:
```suggestion
```

**Review by Hurleveur — APPROVED** [review]
I can't speak for balance but I really like the look and style of that ship, and what it brings to the game. Definitely will use it in the Unfettered campaign shortly.

**Review by warp-core — COMMENTED** [review]
The conversation in the FW bounty mission that the Moonless Night appears in mentions that it is "slow as a sloth."
While the Mammoth is still ordinarily not *fast*, it is considerably faster than the perhaps poorly named "Behemoth (Speedy)" it replaces.
It also mentions the number of heavy lasers on the ship as a notable feature, but this one now only has four Heavy Laster Turrets, which is the same as the normal Behemoth.

File: data/human/free worlds 0 prologue.txt [hunk 17]
```diff
@@ -1068,6 +1068,34 @@ mission "FW Bounty 1"
 
 
 
+ship "Mammoth" "Mammoth (Moonless)"
+	outfits
+		"Modified Blaster Turret" 4
```
Comment (Saugia) [inline data/human/free worlds 0 prologue.txt:1073]:
```suggestion
		"Modified Blaster Turret" 2
```

File: data/human/free worlds 0 prologue.txt [hunk 18]
```diff
@@ -1068,6 +1068,34 @@ mission "FW Bounty 1"
...
+		"Heavy Laser Turret" 4
```
Comment (Saugia) [inline data/human/free worlds 0 prologue.txt:1074]:
```suggestion
		"Heavy Laser Turret" 6
```

File: data/human/free worlds 0 prologue.txt [hunk 19]
```diff
@@ -1068,6 +1068,34 @@ mission "FW Bounty 1"
...
+		
+		"Fission Reactor"
+		"LP144a Battery Pack"
+		"D41-HY Shield Generator"
+		"Large Radar Jammer"
+		"Cooling Ducts"
+		"Laser Rifle" 18
+		
+		"AR120 Reverse Thruster"
```
Comment (Saugia) [inline data/human/free worlds 0 prologue.txt:1083]:
```suggestion
```

File: data/human/free worlds 0 prologue.txt [hunk 20]
```diff
@@ -1068,6 +1068,34 @@ mission "FW Bounty 1"
...
+		"A370 Atomic Thruster"
```
Comment (Saugia) [inline data/human/free worlds 0 prologue.txt:1084]:
```suggestion
		"A250 Atomic Thruster"
```

File: data/human/free worlds 0 prologue.txt [hunk 21]
```diff
@@ -1068,6 +1068,34 @@ mission "FW Bounty 1"
...
+		"A375 Atomic Steering"
+		"Scram Drive"
+
+	turret "Modified Blaster Turret"
+	turret "Modified Blaster Turret"
```
Comment (Saugia) [inline data/human/free worlds 0 prologue.txt:1089]:
```suggestion
	turret "Heavy Laser Turret"
	turret "Heavy Laser Turret"
```

**Review by Saugia — CHANGES_REQUESTED** [review]
Per @warp-core 's review, I believe these changes to fix the balance of the FW one. From 4 Mod Blaster Turrets + 4 Heavy Laser Turrets, to 2 Mod Blaster Turrets + 6 Heavy Laser Turrets. Downgraded the thrusters, and removed reverse thruster to fit them, which does make it slow as molasses as intended.

**Review by Saugia — APPROVED** [review]

File: copyright [hunk 22]
```diff
@@ -907,6 +907,13 @@ Copyright: Michael Zahniser <mzahniser@gmail.com>
 License: CC-BY-SA-4.0
 Comment: Detailed by Becca Tommaso (tommasobecca03@gmail.com).
 
+Files:
+ images/ship/mammoth
+ images/thumbnail/mammoth
```
Comment (Saugia) [inline copyright:912]:
```suggestion
 images/ship/mammoth*
 images/thumbnail/mammoth*
```

File: copyright [hunk 23]
```diff
@@ -907,6 +907,13 @@ Copyright: Michael Zahniser <mzahniser@gmail.com>
...
+Copyright: 1010todd
+License: CC-BY-SA-4.0
+Comment: Derived from works by Michael Zahniser and Becca Tommaso. Incorporating texture made with JSPlacement by WindMillArt <https://windmillart.net/>.
```
Comment (Saugia) [inline copyright:915]:
```suggestion
Comment: Derived from works by Michael Zahniser and Becca Tommaso under the same license. Incorporating texture made with JSPlacement by WindMillArt <https://windmillart.net/>.
```

**Review by Saugia — CHANGES_REQUESTED** [review]
Copyright stuffs.

**Review by Saugia — APPROVED** [review]

**Pull request merged**